universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +33,52 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         executionContext);
   }
 
+  static class PathConflictDetector {
+    private boolean conflict = false;
+    private final String newPath;
+
+    PathConflictDetector(String newPath) {
+      this.newPath = checkNotNull(newPath);
+    }
+
+    public Boolean hasConflict() {
+      return Boolean.valueOf(conflict);
+    }
+
+    public boolean checkConflict(Question question) {
+      boolean proceed = true;
+      if (pathConflicts(question.getPath(), newPath)) {
+        conflict = true;
+        proceed = false;
+      }
+      return proceed;
+    }
+
+    public static boolean pathConflicts(String path, String otherPath) {
+      path = path.toLowerCase();
+      otherPath = otherPath.toLowerCase();
+      return path.startsWith(otherPath) || otherPath.startsWith(path);
+    }
+  }
+
+  public CompletionStage<Boolean> lookupPathConflict(String newPath) {
+    return supplyAsync(
+        () -> {
+          PathConflictDetector detector = new PathConflictDetector(newPath);
+          ebeanServer.find(Question.class).findEachWhile(detector::checkConflict);
```

Comment:
Question (blocking): I would be curious to know how this behaves when there are 100s of questions. Does it generate a separate query for each question? If so it could end up being very slow.

Reply:
You can examine the behavior by turning SQL logging on and running a test that has many questions in the DB.

Reply:
From the test I ran on my local machine with 200+ questions, Ebean only issues one select statement. From the [document](https://ebean.io/docs/query/findEachWhile) this is actually recommended for large queries. For smaller queries `findList` is more performant.

Reply:
Glad to hear it, thanks!